src/components/interpolatedColorLegend.ts

```diff
@@ -197,28 +249,29 @@ export module Component {
                 textRotation: 0
             };
 
+      // Width of each individual swatch box
       var swatchWidth: number;
+      // Height of each individual swatch box
       var swatchHeight: number;
       var swatchX: (d: any, i: number) => number;
       var swatchY: (d: any, i: number) => number;
 
       var boundingBoxAttr = {
         x: 0,
-        y: padding,
+        y: 0,
         width: 0,
         height: 0
       };
 
       if (this._isVertical()) {
-        var longestTextWidth = Math.max(text0Width, text1Width);
-        swatchWidth = Math.max( (this.width() - 3 * padding - longestTextWidth), 0);
-        swatchHeight = Math.max( ((this.height() - 2 * padding) / numSwatches), 0);
-        swatchY = (d: any, i: number) => padding + (numSwatches - (i + 1)) * swatchHeight;
+        var longestTextWidth = Math.max(lowerTextWidth, upperTextWidth);
+        swatchWidth = Math.max(textHeight, 0);
+        swatchHeight = this._isExpanded ? this.height() / numSwatches : swatchWidth;
```

Comment:
The previous logic was really important; it reduced the height of the swatches when the space was constrained. Now you get this:
![screen shot 2015-04-21 at 10 50 34 am](https://cloud.githubusercontent.com/assets/1440449/7258892/42eaf9c8-e814-11e4-9e23-997aabad231c.png)
http://jsfiddle.net/mantxkxx/